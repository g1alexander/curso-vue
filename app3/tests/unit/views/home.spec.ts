import HomeVue from "@/views/Home.vue";
import { shallowMount } from "@vue/test-utils";

describe("pruebas en el Home view", () => {
  test("debe de renderizar el componente correctamente", () => {
    const wrapper = shallowMount(HomeVue);

    expect(wrapper.html()).toMatchSnapshot();
  });

  test("hacer click en un boton debe de redireccionar a no-entry", () => {
    const mockRouter = {
      push: jest.fn(),
    };

    const wrapper = shallowMount(HomeVue, {
      global: {
        mocks: {
          $router: mockRouter,
        },
      },
    });

    wrapper.find("button").trigger("click");

    expect(mockRouter.push).toHaveBeenCalled();
    expect(mockRouter.push).toBeCalledWith({ name: "no-entry" });
  });
});
