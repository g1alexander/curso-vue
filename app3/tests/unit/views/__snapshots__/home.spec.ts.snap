// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en el Home view debe de renderizar el componente correctamente 1`] = `<div class="home"><img alt="Vue logo"><button class="btn btn-dark">primary</button><button class="btn btn-primary">secondary</button><button class="btn btn-secondary">success</button></div>`;
